lib/Transforms/PipelineDataTransfer.cpp

```diff
@@ -309,11 +309,18 @@ void PipelineDataTransfer::runOnAffineForOp(AffineForOp forOp) {
       LLVM_DEBUG(llvm::dbgs() << "tag double buffering failed\n";);
```

Comment:
Out-of-scope for this diff, but FYI there is no coverage in this branch.

Reply:
This will be reached if the tag associated with the DMA transfer escapes in between (almost no use case in practice) - so missed a test case here, but I can add one in this PR; it's very much related to the other changes.

Reply:
Done.